test/mini_racer_test.rb

```diff
@@ -974,51 +848,15 @@ def test_pipe_leak
 
   def test_symbol_support
     context = MiniRacer::Context.new()
-    assert_equal :foo, context.eval("Symbol('foo')")
-    assert_equal :undefined, context.eval("Symbol()") # should not crash
-  end
-
-  def test_cyclical_object_js
-    if RUBY_ENGINE == "truffleruby"
-      skip "TruffleRuby does not yet implement marshal_stack_depth"
-    end
-    context = MiniRacer::Context.new(marshal_stack_depth: 5)
-    context.attach("a", proc { |a| a })
-
-    assert_raises(MiniRacer::RuntimeError) do
-      context.eval("var o={}; o.o=o; a(o)")
-    end
-  end
-
-  def test_cyclical_array_js
-    if RUBY_ENGINE == "truffleruby"
-      skip "TruffleRuby does not yet implement marshal_stack_depth"
-    end
-    context = MiniRacer::Context.new(marshal_stack_depth: 5)
-    context.attach("a", proc { |a| a })
-
-    assert_raises(MiniRacer::RuntimeError) do
-      context.eval("let arr = []; arr.push(arr); a(arr)")
-    end
-  end
-
-  def test_cyclical_elem_in_array_js
-    if RUBY_ENGINE == "truffleruby"
-      skip "TruffleRuby does not yet implement marshal_stack_depth"
-    end
-    context = MiniRacer::Context.new(marshal_stack_depth: 5)
-    context.attach("a", proc { |a| a })
-
-    assert_raises(MiniRacer::RuntimeError) do
-      context.eval("let arr = []; arr[0]=1; arr[1]=arr; a(arr)")
-    end
+    assert_equal "foo", context.eval("Symbol('foo')")
```

Comment:
Is this change intentional?
It seems a not necessary breaking change.

Similar for the line below.

Reply:
Could you somehow serialize it as a string + some marker/metadata and then create a Symbol for it on the Ruby side from that string?

In general, this seems like a bigger difference between this PR and before and truffleruby, the latter two could return any live JS object/value, but only Cloneable is more restrictive. I'm not sure what's a good way to deal with that.

Reply:
It's something I thought about when I worked on it but the short answer is "not easily" because serialization is handled by V8.

I did add a hack for function objects (transformed into strings with hopefully unique prefixes) but I'm really not happy about that so I'd rather not do it twice.

Reply:
I see, thanks for the details.
@SamSaffron What do you think?
Should MiniRacer return Ruby String or Symbol for JS Symbols?
And should the TruffleRuby backend return Ruby String or Symbol for JS Symbols? I can easily do either.

I think this change would be worth adding to the CHANGELOG, along with all other incompatible changes (removal of isolates, some iterators are now eagerly converted to arrays)